tests/test_parameters.py

```diff
@@ -686,3 +688,162 @@ def test_invalid_parameter_values():
     data = {'name': 'my_parameter', 'type': 'AParameterThatShouldHaveValues'}
     with pytest.raises(ValueError):
         load_parameter_values(model, data)
+
+
+class Test1DPolynomialParameter:
+    """ Tests for `Polynomial1DParameter` """
+    def test_init(self, simple_storage_model):
+        """ Test initialisation raises error with too many keywords """
+        stg = simple_storage_model.nodes['Storage']
+
+        with pytest.raises(ValueError):
+            # Passing both "parameter" and "storage_node" is invalid
+            Polynomial1DParameter([0.5, np.pi], parameter=ConstantParameter(2.0), storage_node=stg)
+
+    def test_1st_order_with_parameter(self, model):
+        """ Test 1st order with a `Parameter` """
+        p1 = Polynomial1DParameter([0.5, np.pi], parameter=ConstantParameter(2.0))
+        si = ScenarioIndex(0, np.array([0], dtype=np.int32))
+        ts = model.timestepper.current
+        np.testing.assert_allclose(p1.value(ts, si), 0.5 + np.pi*2.0)
+
+    def test_2nd_order_with_parameter(self, model):
+        """ Test 2nd order with a `Parameter` """
```

Comment:
This test could be made more clear if you used a variable when defining the constant parameter (e.g. `x = 2.0` `p1 = ConstantParameter(x)` ... `0.5 + np.pi*x + 3.0*x**2`). Also, don't redefine p1.